internal/indexer/document_change.go

```diff
@@ -76,25 +76,50 @@ func (idx *Indexer) decodeModule(ctx context.Context, modHandle document.DirHand
 		Type:        op.OpTypeLoadModuleMetadata.String(),
 		DependsOn:   dependsOn,
 		IgnoreState: ignoreState,
-	})
-	if err != nil {
-		return ids, err
-	}
-	ids = append(ids, metaId)
-
-	eSchemaId, err := idx.jobStore.EnqueueJob(ctx, job.Job{
-		Dir: modHandle,
-		Func: func(ctx context.Context) error {
-			return module.PreloadEmbeddedSchema(ctx, idx.logger, schemas.FS, idx.modStore, idx.schemaStore, modHandle.Path())
+		Defer: func(ctx context.Context, jobErr error) (jobIds job.IDs, err error) {
+			if jobErr != nil {
+				err = jobErr
+				return
+			}
+			modCalls, mcErr := idx.decodeDeclaredModuleCalls(ctx, modHandle, ignoreState)
+			if mcErr != nil {
+				idx.logger.Printf("decoding declared module calls for %q failed: %s", modHandle.URI, mcErr)
+				// We log the error but still continue scheduling other jobs
+				// which are still valuable for the rest of the configuration
+				// even if they may not have the data for module calls.
+			}
+
+			eSchemaId, err := idx.jobStore.EnqueueJob(ctx, job.Job{
+				Dir: modHandle,
+				Func: func(ctx context.Context) error {
+					return module.PreloadEmbeddedSchema(ctx, idx.logger, schemas.FS, idx.modStore, idx.schemaStore, modHandle.Path())
```

Comment:
What's the reason behind moving this job into the `Defer` function?

Reply:
Because the origin collection job relies on it - not for collecting the module input origins, but for all the ones which are inside provider-schema-dependent parts of the configuration, i.e. `resource` and `data` blocks.

Reply:
Yes, good catch, I'll add that.

Reply:
Done